DetectorDescription/DDCMS/src/DDFilteredView.cc

```diff
@@ -481,7 +481,8 @@ std::string_view DDFilteredView::get<string_view>(const string& key) const {
       int count = names.size();
       bool flag = false;
       for (int nit = level; count > 0 and nit > 0; --nit) {
-        if (!compareEqual(noNamespace(it_.back().GetNode(nit)->GetVolume()->GetName()), names[--count])) {
+        std::string_view name = noNamespace(it_.back().GetNode(nit)->GetVolume()->GetName());
+        if (!regex_match(std::string(name.data(), name.size()), regex(std::string(names[--count])))) {
```

Comment:
```suggestion
        if (!regex_match(name, regex(std::string(names[--count])))) {
```

Reply:
Suggestion shows if `name` were a `std::string`. I'm missing why it is defined as a `string_view`.

Reply:
@cvuosalo - that’s what the function returns :-)

Reply:
But `name` is used only for `regex_match()`?

Reply:
yes